utils/thorchain-tx-provider.ts

```diff
@@ -0,0 +1,369 @@
+
+  constructor(
+    private chainKey: ChainKey,
+    private chainRef: ChainRef,
+    private dataEncoder: (data: Uint8Array) => Promise<string>,
+    private walletCore: WalletCore
+  ) {
+    this.chainKey = chainKey;
+    this.chainRef = chainRef;
+    this.dataEncoder = dataEncoder;
+    this.walletCore = walletCore;
+
+    this.provider = new JsonRpcProvider(rpcUrl[this.chainKey]);
+  }
+
+  private encryptionKeyHex = (): string => {
+    const keyBytes = randomBytes(32);
+
+    return Array.from(keyBytes)
+      .map((byte) => byte.toString(16).padStart(2, "0"))
+      .join("");
+  };
+
+  private stripHexPrefix = (hex: string): string => {
+    return hex.startsWith("0x") ? hex.slice(2) : hex;
+  };
+
+  public getEstimateTransactionFee = (
+    cmcId: number,
+    currency: Currency
+  ): Promise<string> => {
+    return new Promise((resolve) => {
+      api
+        .cryptoCurrency(cmcId, currency)
+        .then((price) => {
+          const gwei = formatUnits(
+            (this.gasPrice + this.maxPriorityFeePerGas) *
+              BigInt(this.getGasLimit()),
+            "gwei"
+          );
+
+          resolve((parseInt(gwei) * 1e-9 * price).toValueFormat(currency));
+        })
+        .catch(() => {
+          resolve((0).toValueFormat(currency));
+        });
+    });
+  };
+
+  public getSpecificTransactionInfo = (
+    coin: Coin
+  ): Promise<SpecificThorchain> => {
+    return new Promise<SpecificThorchain>((resolve) => {
+      api.thorchain.fetchAccountNumber(coin.address).then((accountData) => {
+        this.calculateFee(coin).then((fee) => {
+          const specificThorchain: SpecificThorchain = {
+            fee,
+            gasPrice: fee,
+            accountNumber: Number(accountData?.accountNumber),
+            sequence: Number(accountData.sequence ?? 0),
+            isDeposit: false,
+          } as SpecificThorchain;
+
+          resolve(specificThorchain);
+        });
+      });
+    });
+  };
+
+  public getGasLimit = (): number => {
+    //TODO: update gaslimit based on chain and transaction type
+    return 600000;
+  };
+
+  public getKeysignPayload = (
+    transaction: TransactionProps,
+    vault: VaultProps
+  ): Promise<KeysignPayload> => {
+    return new Promise((resolve, reject) => {
+      const coin = create(CoinSchema, {
+        chain: transaction.chain.name,
+        ticker: transaction.chain.ticker,
+        address: transaction.from,
+        decimals: transaction.chain.decimals,
+        hexPublicKey: vault.chains.find(
+          (chain) => chain.name === transaction.chain.name
+        ).derivationKey,
+        isNativeToken: true,
+        logo: transaction.chain.ticker.toLowerCase(),
+      });
+      this.getSpecificTransactionInfo(coin).then((specificData) => {
+        const thorchainSpecific = create(THORChainSpecificSchema, {
+          accountNumber: BigInt(specificData.accountNumber),
+          fee: BigInt(specificData.fee),
+          isDeposit: false,
```

Comment:
isDeposit can get from specifData as well

Reply:
I will add deposit in another PR